spec/attribute.dd

```diff
@@ -77,6 +77,7 @@ $(GRAMMAR
 $(GNAME LinkageAttribute):
     $(D extern) $(D $(LPAREN)) $(GLINK LinkageType) $(D $(RPAREN))
     $(D extern) $(D $(LPAREN)) $(D C++), $(GLINK2 declaration, IdentifierList) $(D $(RPAREN))
+    $(D extern) $(D $(LPAREN)) $(D C++), $(GLINK2 declaration, ArgumentList) $(D $(RPAREN))
```

Comment:
I think we may need to introduce a new rule here. The parser looks for a comma-separated list of conditional expressions, while ArgumentList is a comma-separated list of AssignExpression.

Reply:
I'm not convinced. As it is it seems to happily parse an `AssignExpression`. For example `extern(C++, (a = "b")) int c();` parses fine, just like the identifier with an assign index works fine: `extern(C++, (d[e = f])) int g();`

Reply:
Yes, `extern(C++, (a = "b")) int c();` parses, but `extern(C++, a = "b") int c();` does not, which is my point. Putting the assign expression inside the parenthesis makes it match the rule for PrimaryExpression, which causes it to match the rule for ConditionalExpression.

Reply:
Ah you're quite right - this is not my forte. In that case, is the identifier list grammar wrong too? `extern(C++, d[e = f]) int g();` does not parse.